app/src/main/java/com/artemzin/qualitymatters/api/ApiModule.java

```diff
@@ -31,17 +31,13 @@ public ChangeableBaseUrl provideChangeableBaseUrl() {
 
     @Provides @NonNull @Singleton
     public QualityMattersRestApi provideQualityMattersApi(@NonNull OkHttpClient okHttpClient, @NonNull ObjectMapper objectMapper, @NonNull ChangeableBaseUrl changeableBaseUrl) {
-        final Retrofit.Builder builder = new Retrofit.Builder()
+        return new Retrofit.Builder()
                 .baseUrl(changeableBaseUrl)
                 .client(okHttpClient)
                 .addConverterFactory(JacksonConverterFactory.create(objectMapper))
-                .addCallAdapterFactory(RxJavaCallAdapterFactory.create());
-
-        // Fail early: check Retrofit configuration at creation time
-        if (BuildConfig.DEBUG) {
-            builder.validateEagerly();
-        }
-
-        return builder.build().create(QualityMattersRestApi.class);
+                .addCallAdapterFactory(RxJavaCallAdapterFactory.create())
+                .validateEagerly(BuildConfig.DEBUG)  // Fail early: check Retrofit configuration at creation time in Debug build.
```

Comment:
Good idea! The only downside of this is that it'll require providing `Retrofit.Builder`, but seems like worth it